integration/class-perfecty-push.php

```diff
@@ -0,0 +1,329 @@
+		$message = \html_entity_decode( \get_comment_excerpt( $comment ), ENT_QUOTES, 'UTF-8' );
+
+		self::send_notification( $user_id, $message, $title, $actor_image, get_comment_link( $comment ) );
+	}
+
+	/**
+	 * Handle follow notifications.
+	 *
+	 * @param string   $actor_url    The actor URL.
+	 * @param array    $activity     The activity object.
+	 * @param int      $user_id      The user ID.
+	 * @param \WP_Post $remote_actor The remote actor object.
+	 */
+	public static function handle_follow( $actor_url, $activity, $user_id, $remote_actor ) {
+		if ( ! self::is_notification_enabled( 'follow', $user_id ) ) {
+			return;
+		}
+
+		$remote_actor = Remote_Actors::get_actor( $remote_actor );
+		$actor_name   = $remote_actor->get_name() ?? $remote_actor->get_preferred_username();
+		$actor_image  = object_to_uri( $remote_actor->get_icon() ?? ACTIVITYPUB_PLUGIN_URL . 'assets/img/mp.jpg' );
+
+		$title = \__( 'New follower', 'activitypub' );
+		/* translators: %s: Actor name */
+		$message = sprintf( \__( '%s is now following you', 'activitypub' ), $actor_name );
+
+		self::send_notification( $user_id, $message, $title, $actor_image );
+	}
+
+	/**
+	 * Send a push notification to a user.
+	 *
+	 * @param int    $user_id The WordPress user ID.
+	 * @param string $message The notification message.
+	 * @param string $title   The notification title.
+	 * @param string $image_url Optional image URL.
+	 * @param string $url_to_open Optional URL to open when clicked.
+	 */
+	private static function send_notification( $user_id, $message, $title = '', $image_url = '', $url_to_open = '' ) {
+		try {
+			// Check if Perfecty Push Integration class exists and is properly loaded.
+			if ( ! class_exists( 'Perfecty_Push_Integration' ) ) {
+				// Attempt to load the integration file manually.
+				$integration_file = WP_PLUGIN_DIR . '/' . dirname( PERFECTY_PUSH_BASENAME ) . '/integration/class-perfecty-push-integration.php';
+				if ( file_exists( $integration_file ) ) {
+					include_once $integration_file;
+				}
+			}
+
+			if ( class_exists( 'Perfecty_Push_Integration' ) ) {
+				( new \Perfecty_Push_Integration() )->notify( $user_id, $message, $title, $image_url, $url_to_open );
+			}
+		} catch ( \Exception $e ) {
+			error_log( 'ActivityPub Perfecty Push notification failed: ' . $e->getMessage() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+		}
+	}
+
+	/**
+	 * Check if a specific notification type is enabled for a user.
+	 *
+	 * @param string $type The notification type (like, announce, create, follow).
+	 * @param int    $user_id The user ID to check settings for.
+	 * @return bool True if enabled, false otherwise.
+	 */
+	private static function is_notification_enabled( $type, $user_id ) {
+		$default_enabled = array( 'like', 'announce', 'create', 'follow' );
+
+		// Check user-specific settings first.
+		$user_settings = \get_user_meta( $user_id, 'activitypub_perfecty_push_notifications', true );
```

Comment:
notification settings should be generic and plugin independent. It doesn't matter what plugin you use, the settings should be the same.